src/pokemon.c

```diff
@@ -0,0 +1,5567 @@
+#include "struct_decls/struct_0202CC84_decl.h"
+#include "struct_decls/struct_party_decl.h"
+
+#include "struct_defs/struct_02007C10.h"
+#include "struct_defs/struct_02008A90.h"
+#include "struct_defs/struct_0200D0F4.h"
+#include "struct_defs/struct_0202818C.h"
+#include "struct_defs/struct_0202CA28.h"
+#include "struct_defs/pokemon.h"
+#include "struct_defs/box_pokemon.h"
+#include "struct_defs/party_pokemon.h"
+#include "struct_defs/pokemon_data_block_a.h"
+#include "struct_defs/pokemon_data_block_b.h"
+#include "struct_defs/pokemon_data_block_c.h"
+#include "struct_defs/pokemon_data_block_d.h"
+#include "struct_defs/pokemon_personal_data.h"
+#include "struct_defs/pokemon_evolution_data.h"
+#include "struct_defs/pokemon_data_block_shuffle_0.h"
+#include "struct_defs/pokemon_data_block_shuffle_1.h"
+#include "struct_defs/pokemon_data_block_shuffle_10.h"
+#include "struct_defs/pokemon_data_block_shuffle_11.h"
+#include "struct_defs/pokemon_data_block_shuffle_12.h"
+#include "struct_defs/pokemon_data_block_shuffle_13.h"
+#include "struct_defs/pokemon_data_block_shuffle_14.h"
+#include "struct_defs/pokemon_data_block_shuffle_15.h"
+#include "struct_defs/pokemon_data_block_shuffle_16.h"
+#include "struct_defs/pokemon_data_block_shuffle_17.h"
+#include "struct_defs/pokemon_data_block_shuffle_18.h"
+#include "struct_defs/pokemon_data_block_shuffle_19.h"
+#include "struct_defs/pokemon_data_block_shuffle_2.h"
+#include "struct_defs/pokemon_data_block_shuffle_20.h"
+#include "struct_defs/pokemon_data_block_shuffle_21.h"
+#include "struct_defs/pokemon_data_block_shuffle_22.h"
+#include "struct_defs/pokemon_data_block_shuffle_23.h"
+#include "struct_defs/pokemon_data_block_shuffle_3.h"
+#include "struct_defs/pokemon_data_block_shuffle_4.h"
+#include "struct_defs/pokemon_data_block_shuffle_5.h"
+#include "struct_defs/pokemon_data_block_shuffle_6.h"
+#include "struct_defs/pokemon_data_block_shuffle_7.h"
+#include "struct_defs/pokemon_data_block_shuffle_8.h"
+#include "struct_defs/pokemon_data_block_shuffle_9.h"
+#include "struct_defs/struct_020789BC.h"
+#include "struct_defs/struct_020789F4.h"
+#include "struct_defs/struct_02078B40.h"
+#include "overlay005/struct_ov5_021DE5D0.h"
+#include "overlay104/struct_ov104_0223F9E0.h"
+
+#include "unk_020021B0.h"
+#include "unk_02005474.h"
+#include "unk_02006224.h"
+#include "narc.h"
+#include "unk_0200AC5C.h"
+#include "unk_0200B29C.h"
+#include "unk_0200C6E4.h"
+#include "unk_0201378C.h"
+#include "unk_02015F84.h"
+#include "unk_02017038.h"
+#include "heap.h"
+#include "unk_0201D15C.h"
+#include "unk_020218BC.h"
+#include "strbuf.h"
+#include "unk_02025E68.h"
+#include "unk_02028124.h"
+#include "unk_0202C9F4.h"
+#include "pokemon.h"
+#include "move_table.h"
+#include "unk_02079170.h"
+#include "party.h"
+#include "item.h"
+#include "unk_02092494.h"
+
+#include "constants/pokemon.h"
+#include "constants/species.h"
+#include "constants/items.h"
+#include "constants/moves.h"
+
+static const s8 Unk_020F0695[][5] = {
```

Comment:
Looks like it's referenced *only* in `sub_02077648`, which is invoked a bunch of times inside of overlay 16 (which has a bunch of battle code). Those checks look like they're doing stuff with `hpCalcTemp`, and I can see a reference to `207`, which I also know is the battle subroutine sequence for the healing pinch-berries (Figy, Wiki, Iapapa, Mago, and Aguav). I would intuit, then, that since it's nature-related, that these are the flavor-affinities for each nature.